src/db.ts

```diff
@@ -0,0 +1,401 @@
+import fs from "fs";
+
+import sqlite3 from "sqlite3";
+
+const DEFAULT_DB_PATH = "/etc/upstream-syncer/syncer.db";
+
+const createSQLStatements = [
+  // A table to store an upstream or fork repo. It's only useful as a foreign key in an effort of data deduplication.
+  `
+    CREATE TABLE IF NOT EXISTS repos
+        (
+            id INTEGER PRIMARY KEY autoincrement,
+            organization TEXT NOT NULL,
+            name TEXT NOT NULL
+        )
+  `,
+  `
+    CREATE UNIQUE INDEX IF NOT EXISTS idx_repos ON repos (organization, name);
+  `,
+  // A table with a row per upstream repo, fork repo, and fork branch. When such a combination is first encountered
+  // based on where the GitHub app is installed and app's configuration, a "dummy" row is added to the table using the
+  // last PR that was merged for that upstream repo's branch. All future upstream merged PRs for that branch are updated
+  // in that row when properly handled.
+  `
+    CREATE TABLE IF NOT EXISTS last_handled_prs
+        (
+            id INTEGER PRIMARY KEY autoincrement,
+            repo_id INTEGER NOT NULL,
+            branch TEXT NOT NULL,
+            upstream_repo_id INTEGER NOT NULL,
+            upstream_pr_id INTEGER NOT NULL,
+            FOREIGN KEY(repo_id) REFERENCES repos(id)
+            FOREIGN KEY(upstream_repo_id) REFERENCES repos(id)
+        )
+  `,
+  `
+    CREATE UNIQUE INDEX IF NOT EXISTS idx_last_handled_prs ON last_handled_prs (repo_id, upstream_repo_id, branch);
+  `,
+  // A table with a row per upstream repo, fork repo, and fork branch. When a PR is succesfully created on the
+  // fork repo, an entry is added with the "action" column set to "Created". When a PR can't be created on the fork repo
+  // (e.g. merge conflict), an entry is added with action columnt set to "Blocked" and the "github_issue" column set to
+  // the GitHub issue ID that is pending human action.
```

Comment:
I don't know if the right place is here, or on the `pendingPR` type, but I think we could use some doc/comment that a pendingPR may address multiple upstream changes... if I'm reading this right, at least 😅 

That's the idea right? If multiple things get merged upstream, then just one PR will exist downstream, with all the changes? Otherwise it seems like it could run into odd issues with the `setLastHandledPR` logic (again, if I understand it)